### PR TITLE
Clothoidal path planner

### PR DESCRIPTION
**Pull request by daniel-s-ingram**
Hi Atsushi, it's been a little while since my last PR, but I'm hoping to get more involved again! I'll add more detail to the PR soon.

Example outputs:
clothoid
clothoid
clothoid

### CODE REVIEW

**Comment by daniel-s-ingram** [comment]
Latest commit allows for more possibilities like this:
clothoid

**Comment by daniel-s-ingram** [comment]
@AtsushiSakai the six issues found in the `CodeFactor` check also exist in your master branch. Happy to fix them here, but I'm wondering why they're showing up as issues with this PR specifically.

**Comment by AtsushiSakai** [comment]
Maybe that is CodeFactor's issue. So you can ignore these. I think I should remove CodeFacotor check soon.

**Comment by AtsushiSakai** [comment]
Is it ready to review?

**Comment by daniel-s-ingram** [comment]
Yes 👍

File: PathPlanning/ClothoidalPaths/clothoidal_paths.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
+"""
```
Comment (AtsushiSakai) [inline PathPlanning/ClothoidalPaths/clothoidal_paths.py:1]: The directory name and script name should be `clothoidal_pathplanner`?

File: PathPlanning/ClothoidalPaths/clothoidal_paths.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+Clothoidal Path Planner
```
Comment (AtsushiSakai) [inline PathPlanning/ClothoidalPaths/clothoidal_paths.py:2]: Could you please add a unit test file and documentation file based on this doc? :[How to contribute — PythonRobotics documentation](https://pythonrobotics.readthedocs.io/en/latest/how_to_contribute.html#adding-a-new-algorithm-example)

File: PathPlanning/ClothoidalPaths/clothoidal_paths.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+Author: Daniel Ingram (daniel-s-ingram)
+Reference paper: https://www.researchgate.net/publication/237062806
```
Comment (AtsushiSakai) [inline PathPlanning/ClothoidalPaths/clothoidal_paths.py:4]: Could you please rebase this branch to master?

File: PathPlanning/ClothoidalPaths/clothoidal_paths.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+from math import atan2, cos, hypot, pi, sin
+from matplotlib import animation
+
+Point = namedtuple("Point", ["x", "y"])
+
+
+def draw_clothoids(
+    theta1_vals,
+    theta2_vals,
+    num_clothoids=75,
```
Comment (AtsushiSakai) [inline PathPlanning/ClothoidalPaths/clothoidal_paths.py:21]: This argument is no used?

File: PathPlanning/ClothoidalPaths/clothoidal_paths.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    y_min = y_min - y_offset
+    y_max = y_max + y_offset
+
+    return x_min, x_max, y_min, y_max
+
+
+if __name__ == "__main__":
+    theta1_vals = [0]
+    theta2_vals = np.linspace(-pi, pi, 75)
+    draw_clothoids(theta1_vals, theta2_vals, save_animation=False)
```
Comment (AtsushiSakai) [inline PathPlanning/ClothoidalPaths/clothoidal_paths.py:144]: Could you please add `main()` function including these 3 lines and call it here?

File: PathPlanning/ClothoidalPaths/clothoidal_paths.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+def X(a, b, c):
+    return integrate.quad(lambda t: cos((a/2)*t**2 + b*t + c), 0, 1)[0]
+
+
+def Y(a, b, c):
+    return integrate.quad(lambda t: sin((a/2)*t**2 + b*t + c), 0, 1)[0]
+
+
+def solve_for_root(theta1, theta2, delta):
```
Comment (AtsushiSakai) [inline PathPlanning/ClothoidalPaths/clothoidal_paths.py:100]: Can you add docsting for each function to describe what the function is doing and each argument meanings.

File: PathPlanning/ClothoidalPaths/clothoidal_paths.py (hunk 3 above)
Comment (AtsushiSakai) [inline PathPlanning/ClothoidalPaths/clothoidal_paths.py:4]:
```suggestion
Reference paper: Fast and accurate G1 fitting of clothoid curves https://www.researchgate.net/publication/237062806
```

**Review by AtsushiSakai — CHANGES_REQUESTED** [review]
Sorry for waiting for review. I have some comments. PTAL.

**Comment by AtsushiSakai** [comment]
@daniel-s-ingram Do you have time to finish this PR? If not, I can finish it, is it OK for you?

**Comment by AtsushiSakai** [comment]
OK. Let's merge this. Then clean up in another PR. Thanks, @daniel-s-ingram.

**Pull request merged**